crates/bevy_reflect/bevy_reflect_derive/src/registration.rs

```diff
@@ -14,14 +14,44 @@ pub(crate) fn impl_get_type_registration(
     registration_data: &[Ident],
     generics: &Generics,
     where_clause_options: &WhereClauseOptions,
-    serialization_denylist: Option<&BitSet<u32>>,
+    serialization_denylist: Option<&SerializationDenylist>,
 ) -> proc_macro2::TokenStream {
     let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
-    let serialization_data = serialization_denylist.map(|denylist| {
-        let denylist = denylist.into_iter();
-        quote! {
-            let ignored_indices = ::core::iter::IntoIterator::into_iter([#(#denylist),*]);
-            registration.insert::<#bevy_reflect_path::serde::SerializationData>(#bevy_reflect_path::serde::SerializationData::new(ignored_indices));
+
+    let serialization_data = serialization_denylist.map(|denylist| match denylist {
+        SerializationDenylist::Struct(denylist) => {
+            let denylist = denylist.into_iter();
+            quote! {
+                let ignored_indices = ::core::iter::IntoIterator::into_iter([#(#denylist),*]);
+                registration.insert(
+                    #bevy_reflect_path::serde::SerializationData::Struct(
+                        #bevy_reflect_path::serde::StructSerializationData::new(ignored_indices)
+                    )
+                );
+            }
+        }
+        SerializationDenylist::Enum(denylist) => {
+            let mut indices = proc_macro2::TokenStream::new();
```

Comment:
Could be good to explicit that this is a list of tuples `(name, indices)`, as meant for `EnumSerializationData::new` (for example through a better name or a small comment line).

I had difficulties finding what `#indices` was at line 48.